src/hermes/model/types/ld_list.py

```diff
@@ -44,15 +29,7 @@ def __getitem__(self, index):
     def __setitem__(self, index, value):
         # FIXME: #439 what should your_ld_list[index] = [{"@type": "foo", "name": "bar"}] mean?
         # set your_ld_list[index] to the dict {"@type": "foo", "name": "bar"} given in expanded form        or
-        # set your_ld_list[index] to the list [{"@type": "foo", "name": "bar"}] given in non expanded form  or
-        # set your_ld_list[index] to the set [{"@type": "foo", "name": "bar"}] given in expanded form
-        #   (ld_list.fromlist([{"@type": "foo", "name": "bar"}]) defaults to container type list
-        #    which would have the object as an expanded form whereas the expanded form of a list would be
-        #    ["@list": [{"@type": "foo", "name": "bar"}]]
-        #    This is relevent because nested sets get unnested when being expanded and lists not.
-        #    Moreover a set inside a list gets automaticaly converted to a list when expanded)
-
-        # FIXME: #439 what happens when a ld_list is put inside another also depends on their container types
+        # set your_ld_list[index] to the list [{"@type": "foo", "name": "bar"}] given in non expanded form
```

Comment:
Well, it should be considered whether this operation should be allowed in this context. However, in general, it should behave as expected (i.e., set the value to what is passed). What it clearly must not do is setting something in the `data` property that is not valid expanded JSON. People who really want to do this still can directly access item_list or data...

For this concrete question I see two valid solutions: Do handling as to what is easiest implementable (and document) or if in doubt, raise an error.